media-libs/opencv/opencv-3.4.0.recipe

```diff
@@ -241,6 +241,8 @@ INSTALL()
 	# fixCmake
 	sed -i 's,\/include,/'${relativeIncludeDir}',g' \
 		$libDir/cmake/OpenCV/OpenCVConfig.cmake
+	sed -i 's,\/..\/..\/,\/..\/..\/..\/,g' \
```

Comment:
If you are trying to replace `/../../` by `/../../../` then this is not the correct command.
What would do this is:
```
sed -i 's,/\.\./\.\./,/../../,g' \
```
Remember: in sed, a _dot_ means _any character_. If you really want a dot you need to escape it (on the left side of the substitution. On the righ side, a dot is a dot.

Reply:
My solution worked great, i see no point to changing it, as i fear i break it again.

Reply:
The way the current sed command is written, what it does is that it looks for `/ab/cd/` with a, b, c and d being **any** characters, and then it replaces this match by `/../../../`.
So if you are lucky it will do what you are expecting it to do.
But if you are unlucky and there happens to be a match where any (or some, or all) of a, b, c or d are **not** a dot, then you will transform these non dot characters into dots.

BTW, since the sed substitution is using a comma as separator instead of the traditional slash, the slashes in the regexp don't need to be escaped. (What needs to be escaped here are the dots.)

With this release it looks like there are no such matches, but for future releases there could very well be a match. Imagine a `/../ET/` appears in a future release of the file you are patching. You would get `/../../../` and the `ET` would be gone.

You can safely change your sed command to:
```
sed -i -e "s|/\.\./\.\./|&../|g'
```
and it will do exactly what you want: add a `../` everytime you see a `/../../`.
If you try this I'm quite sure you will get the same result as with your current sed command.
The difference is that with this command your are much safer.

But if you wish to keep the sed command the way it is, that's OK for me. (I"m not using opencv on Haiku :)

Reply:
Your example starts with " and ends with '
Which one should i use then?

Thanks for the explanation, you have right.

Reply:
Oops, good catch.
Both work (if fixed to make the quotes match.)
(If we had to use e.g. $dataDir (or any environment variable) then the only correct answer would be with the double quotes, since simple quotes prevent evaluation of environment variables.)